src/components/FormElements/XdrLedgerKeyPicker.tsx

```diff
@@ -0,0 +1,474 @@
+  onChange: (val: string[]) => void;
+  value: string[];
+  buttonLabel?: string;
+  limit?: number;
+};
+
+type XdrLedgerKeyPicker = {
+  value: string;
+  onChange: (value: string) => void;
+  rightElement: ReactElement | null;
+};
+
+// https://github.com/stellar/stellar-xdr/blob/curr/Stellar-ledger-entries.x#L600
+const ledgerKeyFields: {
+  id: LedgerKeyType;
+  label: string;
+  templates: string;
+  custom?: AnyObject;
+}[] = [
+  {
+    id: "account",
+    label: "Account",
+    templates: "account_id",
+  },
+  {
+    id: "trustline",
+    label: "Trustline",
+    templates: "account_id,asset",
+    custom: {
+      assetInput: "alphanumeric",
+      includePoolShare: true,
+    },
+  },
+  {
+    id: "offer",
+    label: "Offer",
+    templates: "seller_id,offer_id",
+  },
+  {
+    id: "data",
+    label: "Data",
+    templates: "account_id,data_name",
+  },
+  {
+    id: "claimable_balance",
+    label: "Claimable Balance",
+    templates: "balance_id",
+  },
+  {
+    id: "liquidity_pool",
+    label: "Liquidity Pool",
+    templates: "liquidity_pool_id",
+  },
+  {
+    id: "contract_data",
+    label: "Contract Data",
+    templates: "contract,key,durability",
+  },
+  {
+    id: "contract_code",
+    label: "Contract Code",
+    templates: "hash",
+  },
+  {
+    id: "config_setting",
+    label: "Config Setting",
+    templates: "config_setting_id",
+  },
+  {
+    id: "ttl",
+    label: "TTL",
+    templates: "key_hash",
+  },
+];
+
+export const XdrLedgerKeyPicker = ({
+  value,
+  onChange,
+  rightElement,
+}: XdrLedgerKeyPicker) => {
+  const [selectedLedgerKey, setSelectedLedgerKey] =
+    useState<LedgerKeyFieldsType | null>(null);
+
+  const [formError, setFormError] = useState<AnyObject>({});
+  const [ledgerKeyXdrJsonString, setLedgerKeyJsonString] = useState<string>("");
+  const [ledgerKeyXdrError, setLedgerKeyXdrError] = useState<string>("");
+
+  const isXdrInit = useIsXdrInit();
+
+  const xdrDecodeJson = (xdr: string) => {
+    if (!isXdrInit) {
+      return null;
+    }
+
+    try {
+      const xdrJson = StellarXdr.decode("LedgerKey", xdr);
+
+      return {
+        jsonString: xdrJson,
+        error: "",
+      };
+    } catch (e) {
+      return {
+        jsonString: "",
+        error: `Unable to decode input as LedgerKey: ${e}`,
+      };
+    }
+  };
+
+  const jsonEncodeXdr = (str: string) => {
+    if (!(isXdrInit && str && "LedgerKey")) {
+      return null;
+    }
+
+    try {
+      JSON.parse(str);
+    } catch (e) {
+      return {
+        xdrString: "",
+        error: "Invalid JSON",
+      };
+    }
+
+    try {
+      const xdrString = StellarXdr.encode("LedgerKey", str);
+
+      return {
+        xdrString,
+        error: "",
+      };
+    } catch (e) {
+      return {
+        xdrString: "",
+        error: `Unable to decode JSON as LedgerKey: ${e}`,
+      };
+    }
+  };
+
+  const reset = () => {
+    setFormError({});
+    setLedgerKeyJsonString("");
+    onChange("");
+  };
```

Comment:
Should we reset error as well?

```suggestion
  const reset = () => {
    setFormError({});
    setLedgerKeyJsonString("");
    setLedgerKeyXdrError("")
    onChange("");
  };
```